### PR TITLE
Add support to specify border-radius around labels

### PR DESCRIPTION
**Pull request by shashankshukla96**
Associated issues: #3127  #3093 

- This PR adds a feature to specify border-radius around node/edge labels
- This PR fixes text-border overrides text-background-shape

**Checklist**

Author:

- [x] The proper base branch has been selected.  New features go on `unstable`.  Bug-fix patches can go on either `unstable` or `master`.
- [x] Automated tests have been included in this pull request, if possible, for the new feature(s) or bug fix.  Check this box if tests are not pragmatically possible (e.g. rendering features could include screenshots or videos instead of automated tests).
- [x] The associated GitHub issues are included (above).
- [x] Notes have been included (above).

Reviewers:

- [ ] All automated checks are passing (green check next to latest commit).
- [ ] At least one reviewer has signed off on the pull request.
- [ ] For bug fixes:  Just after this pull request is merged, it should be applied to both the `master` branch and the `unstable` branch.  Normally, this just requires cherry-picking the corresponding merge commit from `master` to `unstable` -- or vice versa.

### CODE REVIEW

File: documentation/md/style.md [hunk 1]
```diff
@@ -619,6 +619,7 @@ Border:
  * **`text-border-width`** : The width of the border around the label.
  * **`text-border-style`** : The style of the border around the label; may be `solid`, `dotted`, `dashed`, or `double`.
  * **`text-border-color`** : The colour of the border around the label.
+ * **`text-border-radius`**: The radius of the border around the label.
```
Comment (maxkfranz) [inline documentation/md/style.md:622]: Will this work with all `text-background-shape`s, or will it only work with particular ones?

File: src/extensions/renderer/canvas/drawing-label-text.js [hunk 2]
```diff
@@ -164,38 +164,38 @@ CRp.getTextAngle = function( ele, prefix ){
...
+  let textX = util.getPrefixedProperty(rscratch, 'labelX', prefix);
+  let textY = util.getPrefixedProperty(rscratch, 'labelY', prefix);
   let orgTextX, orgTextY; // used for rotation
-  let text = this.getLabelText( ele, prefix );
+  let text = this.getLabelText(ele, prefix);
+
+  if (text != null && text !== '' && !isNaN(textX) && !isNaN(textY)) {
 
-  if( text != null && text !== '' && !isNaN( textX ) && !isNaN( textY ) ){
-    this.setupTextStyle( context, ele, useEleOpacity );
```
Comment (maxkfranz) [inline src/extensions/renderer/canvas/drawing-label-text.js:185]: Would you please revert any non-functional, formatting-related changes in your PR?  You're welcome to make formatting changes, but they should be made in a separate PR which includes only formatting changes.

It's not tractable to review your proposed changes with a noisy diff.

**Review by maxkfranz — CHANGES_REQUESTED** [review]
This looks like a great PR, and we'd be happy to incorporate this feature into the next release of the library.

Rendering is generally difficult, and it's not easy to test in an automated way like much of the core API.  So, it's important that we are able to make at least a best-effort review of your changes to double-check for possible unintended side-effects or bugs outside of your happy-path.

In order to carry out the review, we need to be able to look at the changes meaningfully and that's not tractable with all the unrelated formatting changes.

We're looking forward to your updates, and again, we'd like to get this in the next release.

**Comment by maxkfranz** [comment]
@shashankshukla96, do you have any questions re. the review comments?

File: documentation/md/style.md (hunk 1 above)
Comment (shashankshukla96) [reply documentation/md/style.md:622]: if a text-border-radius is set, it will override the text-background-shape value

**Comment by maxkfranz** [comment]
Thanks for the updates, @shashankshukla96.  I'll try to set aside some time for review later this week.

File: documentation/md/style.md [hunk 3]
```diff
@@ -619,6 +619,7 @@ Border:
  * **`text-border-width`** : The width of the border around the label.
  * **`text-border-style`** : The style of the border around the label; may be `solid`, `dotted`, `dashed`, or `double`.
  * **`text-border-color`** : The colour of the border around the label.
+ * **`text-border-radius`**: The radius of the border around the label. This will override the `text-background-shape` property.
```
Comment (maxkfranz) [inline documentation/md/style.md:622]: It would be best to restrict the application of `text-border-radius` to `text-background-shape: round-rectangle`.    Otherwise, it will introduce inconsistencies in future.

**Review by maxkfranz — CHANGES_REQUESTED** [review]
This looks like a good start.

(1) See the comment re. the property behaviour.

(2) The feature does not seem to function properly.  Try the debug page when you run `npm run watch`.  You just need to set the style appropriately in `init.js`:

```
.selector('edge')
        .style({
          'source-arrow-shape': 'triangle-backcurve',
          'target-arrow-shape': 'triangle',
          'mid-target-arrow-shape': 'triangle',
          'mid-source-arrow-shape': 'triangle-backcurve',
          'label': 'my test label',
          'text-background-shape': 'round-rectangle',
          'text-background-color': '#ddd',
          'text-background-opacity': 1,
          'text-background-padding': 10,
          // 'text-border-radius': 10 // broken
        })
```

This is how it looks when the custom border radius is not applied to the `round-rectangle` shape:

Screenshot 2023-07-28 at 11 14 58

This is how it breaks when the radius is applied:

Screenshot 2023-07-28 at 11 15 46

The background is broken, and the label is mis-coloured.

There is an issue with the demo you included with your PR:

Screenshot 2023-07-28 at 11 18 16

We're planning on releasing 3.26.0 on next week's Friday, the 4th of August.  It would be great to have this feature ready to merge in before then so that it makes it in the release.

**Comment by maxkfranz** [comment]
@shashankshukla96, would you be able to address the above points prior to Friday, so that this feature can make it in the upcoming release?

File: documentation/md/style.md (hunk 1 above)
Comment (maxkfranz) [reply documentation/md/style.md]: That behaviour wouldn’t be consistent with existing style properties.

**Comment by maxkfranz** [comment]
@shashankshukla96, let us know if any of the feedback is unclear.  

Looking forward to your updates

**Comment by shashankshukla96** [comment]
@maxkfranz sorry I was busy with some work, so could not get a chance to resolve this, I have checked the comments and also made some changes to fix them, I will push them by the end of this week.

**Comment by maxkfranz** [comment]
@shashankshukla96, great. If you want any feedback, just @-mention me.

**Comment by maxkfranz** [comment]
@shashankshukla96, are you still interested in this feature?
